src/libs/actions/Policy.ts

```diff
@@ -309,106 +327,128 @@ function removeOptimisticAnnounceRoomMembers(policyID, accountIDs) {
+        optimisticMembersState[accountID] = {pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE};
+        successMembersState[accountID] = null;
+        failureMembersState[accountID] = {errors: ErrorUtils.getMicroSecondOnyxError('workspace.people.error.genericRemove')};
+    });
+
+    const optimisticData: OnyxUpdate[] = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
             key: membersListKey,
-            value: _.object(accountIDs, Array(accountIDs.length).fill({pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE})),
+            value: optimisticMembersState,
         },
-        ..._.map(workspaceChats, (report) => ({
+        ...announceRoomMembers.onyxOptimisticData,
+    ];
+
+    workspaceChats.forEach((report) => {
+        optimisticData.push({
             onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${report?.reportID}`,
             value: {
                 statusNum: CONST.REPORT.STATUS.CLOSED,
                 stateNum: CONST.REPORT.STATE_NUM.SUBMITTED,
                 oldPolicyName: policy.name,
                 hasDraft: false,
             },
-        })),
-        ..._.map(optimisticClosedReportActions, (reportAction, index) => ({
+        });
+    });
+    optimisticClosedReportActions.forEach((reportAction, index) => {
+        optimisticData.push({
             onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${workspaceChats[index].reportID}`,
-            value: {[reportAction.reportActionID]: reportAction},
-        })),
-        ...announceRoomMembers.onyxOptimisticData,
-    ];
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${workspaceChats?.[index]?.reportID}`,
+            value: {[reportAction.reportActionID]: reportAction as ReportAction},
+        });
+    });
 
     // If the policy has primaryLoginsInvited, then it displays informative messages on the members page about which primary logins were added by secondary logins.
     // If we delete all these logins then we should clear the informative messages since they are no longer relevant.
-    if (!_.isEmpty(policy.primaryLoginsInvited)) {
+    if (isNotEmptyObject(policy?.primaryLoginsInvited ?? {})) {
         // Take the current policy members and remove them optimistically
-        const policyMemberAccountIDs = _.map(allPolicyMembers[`${ONYXKEYS.COLLECTION.POLICY_MEMBERS}${policyID}`], (value, key) => Number(key));
-        const remainingMemberAccountIDs = _.difference(policyMemberAccountIDs, accountIDs);
-        const remainingLogins = PersonalDetailsUtils.getLoginsByAccountIDs(remainingMemberAccountIDs);
-        const invitedPrimaryToSecondaryLogins = _.invert(policy.primaryLoginsInvited);
+        const policyMemberAccountIDs = Object.keys(allPolicyMembers?.[`${ONYXKEYS.COLLECTION.POLICY_MEMBERS}${policyID}`] ?? {}).map((accountID) => parseInt(accountID, 10));
```

Comment:
Why `Number(key)` was replaced with `parseInt(accountID, 10)`?

Reply:
I took over this task from another person so I wasn't sure at first, but after analyzing it, we will go back to using Number here.